phi/tools/calcom.py

```diff
@@ -0,0 +1,235 @@
+from datetime import datetime
+from os import getenv
+from typing import Optional, Dict
+from phi.tools import Toolkit
+from phi.utils.log import logger
+
+try:
+    import requests
+    import pytz
+except ImportError:
+    raise ImportError("`requests` and `pytz` not installed. Please install using `pip install requests pytz`")
+
+
+class CalCom(Toolkit):
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        event_type_id: Optional[int] = None,
+        user_timezone: Optional[str] = None,
+    ):
+        """Initialize the Cal.com toolkit.
+
+        Args:
+            api_key: Cal.com API key
+            event_type_id: Default event type ID for bookings
+            user_timezone: User's timezone in IANA format (e.g., 'Asia/Kolkata')
+        """
+        super().__init__(name="calcom")
+
+        # Get credentials from environment if not provided
+        self.api_key = api_key or getenv("CALCOM_API_KEY")
+        event_type_str = getenv("CALCOM_EVENT_TYPE_ID")
+        self.event_type_id = event_type_id or int(event_type_str) if event_type_str is not None else 0
+
+        if not self.api_key:
+            logger.error("CALCOM_API_KEY not set. Please set the CALCOM_API_KEY environment variable.")
+        if not self.event_type_id:
+            logger.error("CALCOM_EVENT_TYPE_ID not set. Please set the CALCOM_EVENT_TYPE_ID environment variable.")
+
+        self.user_timezone = user_timezone or "America/New_York"
+
+        # Register all methods
+        self.register(self.get_available_slots)
+        self.register(self.create_booking)
+        self.register(self.get_upcoming_bookings)
+        self.register(self.reschedule_booking)
+        self.register(self.cancel_booking)
```

Comment:
add an if condition to let the user decide which functions to allow and which not to. Take example from other tools example slack

Reply:
Thanks @manthanguptaa I've updated code. please let me know if there are any more changes required